src/apps/apollox/binance-smart-chain/apollox.alp.contract-position-fetcher.ts

```diff
@@ -0,0 +1,64 @@
+import { Inject } from '@nestjs/common';
+import { BigNumberish, Contract } from 'ethers';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import { DefaultDataProps } from '~position/display.interface';
+import { MetaType } from '~position/position.interface';
+import { ContractPositionTemplatePositionFetcher } from '~position/template/contract-position.template.position-fetcher';
+import {
+  GetDefinitionsParams,
+  DefaultContractPositionDefinition,
+  GetTokenDefinitionsParams,
+  UnderlyingTokenDefinition,
+  GetDisplayPropsParams,
+  GetTokenBalancesParams,
+} from '~position/template/contract-position.template.types';
+
+import { Alp, ApolloxContractFactory } from '../contracts';
+
+const ALP_CONTRACT = '0x4e47057f45adf24ba41375a175da0357cb3480e5';
+const APX_CONTRACT = '0x78f5d389f5cdccfc41594abab4b0ed02f31398b3';
+
+export type AlpSupplyDetails = DefaultContractPositionDefinition & {
+  apy?: number;
+  TotalAmountStaked?: number;
+  TotalRewards?: number;
+};
+
+@PositionTemplate()
+export class BinanceSmartChainApolloxAlpContractPositionFetcher extends ContractPositionTemplatePositionFetcher<Contract> {
```

Comment:
You will need to replace `Contract` with the interface (`Alp`) you want to use to make calls.